src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java

```diff
@@ -303,6 +303,32 @@ public String getCollapseConstructorQuery(CollapseSettings collapseSettings) {
 		query = StringUtils.replace(query, "@eraconstructorpad", Integer.toString(collapseSettings.eraPad));
 		return query;
   }
+
+  public String getFinalCohortQuery(Period censorWindow) {
+
+    String query = "select @target_cohort_id as cohort_definition_id, person_id, @start_date, @end_date \n" +
+            "FROM #final_cohort CO";
+
+    String startDate = "start_date";
+    String endDate = "end_date";
+
+    if (censorWindow != null && (censorWindow.startDate != null || censorWindow.endDate != null)) {
+      if (censorWindow.startDate != null) {
+        String censorStartDate = dateStringToSql(censorWindow.startDate);
+        startDate = "CASE WHEN start_date > " + censorStartDate + " THEN start_date ELSE " + censorStartDate + " END";
+      }
+      if (censorWindow.endDate != null) {
+        String censorEndDate = dateStringToSql(censorWindow.endDate);
+        endDate = "CASE WHEN end_date < " + censorEndDate + " THEN end_date ELSE " + censorEndDate + " END";
+      }
+      query += "\nWHERE @start_date <= @end_date";
```

Comment:
This is an elegant way to drop people who aren't bounded by the study window, nice!